packages/mobile/src/components/countdown/countdown.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { FC } from 'react';
+import { View } from '~/components/components';
+import { globalStyles } from '~/styles/styles';
+import { useTranslation, useEffect, useState } from '~/hooks/hooks';
+import { Box } from './components/components';
+import { BoxType } from './common/enums/enums';
+import { styles } from './styles';
+import { intervalSetter } from './helpers/interval-setter';
+
+type Props = {
+  endDate: Date;
+};
+
+const Countdown: FC<Props> = ({ endDate }) => {
+  const { t } = useTranslation();
+  const [duration, setDuration] = useState({
+    days: '0',
+    hours: '00',
+    minutes: '00',
+    seconds: '00',
+  });
+
+  useEffect(() => {
+    const timerId = intervalSetter(setDuration, endDate);
+
+    return () => clearInterval(timerId);
+  }, []);
+  const { days, hours, minutes, seconds } = duration;
```

Comment:
Given that you still destructure it, why not store all the values as a separate states?

Reply:
In that case, I will need to pass 4 different setStates  functions to the interval Setter

Reply:
moved  to hook